schematic/store/synapse.py

```diff
@@ -81,31 +85,93 @@ class ManifestDownload(object):
 
     syn: synapseclient.Synapse
     manifest_id: str
+    synapse_entity_tracker: SynapseEntiyTracker = field(
+        default_factory=SynapseEntiyTracker
+    )
 
-    def _download_manifest_to_folder(self) -> File:
+    def _download_manifest_to_folder(self, use_temporary_folder: bool = True) -> File:
         """
-        try downloading a manifest to local cache or a given folder
-        manifest
+        Try downloading a manifest to a specific folder (temporary or not). When the
+        `use_temporary_folder` is set to True, the manifest will be downloaded to a
+        temporary folder. This is useful for when the code is running as an API server
+        where multiple requests are being made at the same time. This will prevent
+        multiple requests from overwriting the same manifest file. When the
+        `use_temporary_folder` is set to False, the manifest will be downloaded to the
+        default manifest folder.
+
+        Args:
+            use_temporary_folder: boolean argument indicating if a temporary folder
+                should be used to store the manifest file. This is useful when running
+                this code as an API server where multiple requests could be made at the
+                same time. This is set to False when the code is being used from the
+                CLI. Defaults to True.
+
         Return:
             manifest_data: A Synapse file entity of the downloaded manifest
         """
+        manifest_data = self.synapse_entity_tracker.get(
+            synapse_id=self.manifest_id,
+            syn=self.syn,
+            download_file=False,
+            retrieve_if_not_present=False,
+        )
+
+        if manifest_data and manifest_data.path:
+            return manifest_data
+
         if "SECRETS_MANAGER_SECRETS" in os.environ:
             temporary_manifest_storage = "/var/tmp/temp_manifest_download"
-            # clear out all the existing manifests
-            if os.path.exists(temporary_manifest_storage):
-                shutil.rmtree(temporary_manifest_storage)
+            cleanup_temporary_storage(
+                temporary_manifest_storage, time_delta_seconds=3600
+            )
```

Comment:
Instead of blanket removing previously downloaded manifest files instead this will have a 1 hour delay to clean them up. This was implemented to prevent a manifest from getting cleaned up that might be actively being used, or used for a successful cache pull (ie: Not downloading the manifest file again).